packages/generator/src/v1.1/generateActions.ts

```diff
@@ -43,49 +43,71 @@ export async function generateActions(
       jsDocBlock.push(`* ${action.description}`);
     }
 
-    let parameterBlock = "";
+    let interiorParameters = "";
     if (entries.length > 0) {
-      parameterBlock = `params: { \n`;
+      interiorParameters = "{ \n";
       for (
         const [parameterName, parameterData] of entries
       ) {
-        parameterBlock += `"${parameterName}"`;
-        parameterBlock += parameterData.required ? ": " : "?: ";
+        interiorParameters += `"${parameterName}"`;
+        interiorParameters += parameterData.required ? ": " : "?: ";
         const typeScriptType = getTypeScriptTypeFromDataType(
           parameterData.dataType,
           importedObjects,
         );
-        parameterBlock += `${typeScriptType};\n`;
+        interiorParameters += `${typeScriptType};\n`;
 
         jsDocBlock.push(
           `* @param {${typeScriptType}} params.${parameterName}`,
         );
       }
-      parameterBlock += "}, ";
+      interiorParameters += "} ";
     }
 
     jsDocBlock.push(`*/`);
     actionSignatures.push(
-      `
+      entries.length > 0
+        ? `
       ${jsDocBlock.join("\n")}
-      ${action.apiName}<O extends ActionExecutionOptions>(${parameterBlock}options?: O): 
-        Promise<Result<ActionResponseFromOptions<O, Edits<${
-        addedObjects.length > 0
-          ? addedObjects.join(" | ")
-          : "void"
-      }, ${
-        modifiedObjects.length > 0
-          ? modifiedObjects.join(" | ")
-          : "void"
-      }>>, ActionError>>;`,
+      ${action.apiName}<P extends ${interiorParameters} | ${interiorParameters}[],O extends P extends ${interiorParameters}[]? BulkActionExecutionOptions: ActionExecutionOptions>(
```

Comment:
So, I made the O generic something like `O extends BulkActionExecutionOptions | ActionExecutionOptions`, which it seemed to like, but I wasn't able to test if that would still enforce our internal argument constraints because I got blocked on another type issue below